include/tvm/relay/vm/vm.h

```diff
@@ -44,45 +44,74 @@ struct Instruction {
   };
 
   Opcode op;
+
+  // Destination register that the opcode writes to
+  VirtualRegisterNum dst;
+
   union {
     size_t stack_index;
     TensorInfo tensor_info;
+
+    // For InvokeClosure
+    struct {
+      VirtualRegisterNum closure;
+      size_t closure_args_num;
+      VirtualRegisterNum* closure_args;
+    };
+    // For Ret
+    struct {
+      VirtualRegisterNum result;
+    };
+    // For Move
+    struct {
+      VirtualRegisterNum from;
+    };
     struct {
       size_t packed_index;
       size_t arity;
       size_t output_size;
+      VirtualRegisterNum* packed_args;
+    };
+    // For Phi node
+    struct {
+      VirtualRegisterNum phi_cond;
+      VirtualRegisterNum phi_op1;
+      VirtualRegisterNum phi_op2;
     };
+    // For If node
     struct {
+      VirtualRegisterNum if_cond;
       size_t true_offset;
       size_t false_offset;
     };
+    // For Invoke
     struct {
       size_t func_index;
+      size_t num_args;
+      VirtualRegisterNum* invoke_args_registers;
     };
     struct {
       size_t const_index;
     };
     struct {
       size_t pc_offset;
     };
+    // For GetField
     struct {
-      size_t object_offset;
+      VirtualRegisterNum object;
       size_t field_index;
     };
+    // For AllocDatatype
     struct {
       size_t constructor_tag;
       size_t num_fields;
+      VirtualRegisterNum* datatype_fields;
     };
+    // For AllocClosure
     struct {
       size_t clo_index;
```

Comment:
Shall we just call it `func_index`, since we don't distinguish closure or vm function in the module, right?